casa/Json/JsonParser.cc

```diff
@@ -125,8 +125,48 @@ namespace casacore {
     for (String::size_type i=0; i<leng; ++i) {
       if (in[i] == '\\') {
         i++;
-      }
-      if (i < leng) {
+        if (i < leng) {
+          switch (in[i]) {
+          case 'b':
+            out += '\b';  // backspace
+            break;
+          case 'f':
+            out += '\f';  // formfeed
+            break;
+          case 'n':
+            out += '\n';  // newline
+            break;
+          case 'r':
+            out += '\r';  // carriage return
+            break;
+          case 't':
+            out += '\t';  // tab
+            break;
+          case 'u':
+            {
+              // unicode repr of control character
+              Bool ok = False;
+              if (i < leng+4) {
+                String hex("0X" + in.substr(i+1,4));
+                char* endPtr;
+                Int64 val = strtoll(hex.chars(), &endPtr, 0);
+                if (endPtr == hex.chars()+hex.size()  ||  val < 128) {
```

Comment:
If hex has the value "0X12hi", the first part of this conditional is false, but val has value 12, and so the second part of the conditional is true. Shouldn't the conditions be combined with && instead?